Use versions from `grails-bom` in example apps

matrei opened the pull request:
And remove unused versions from `gradle.properties`

matrei committed:
fix(deps): Use versions from `grails-bom` in example apps

And remove unused versions from `gradle.properties`

matrei committed:
build: Remove `webdriver-binaries-gradle-plugin`

As `ContainerGebSpec` is used now, `webdriver-binaries-gradle-plugin` can be removed.

jdaugherty commented on gradle.properties:
```diff
@@ -1,9 +1,6 @@
 projectVersion=9.0.0-SNAPSHOT
 #4.0.2-4.0.3 breaks org.ysb33r.gradle:grolifant for chromedriver
 asciidoctorGradleVersion=4.0.1
```
Grails-data-mapping & gorm-hibernate5 are meant to be used outside of Grails applications.  Are any of these versions still required to set version properties correctly when grails-bom isn't in use?  

I'm assuming the jakarta version due to the older hibernate import may be the only one?  The rest look like tests or grails specific ones.

matrei replied:
Yes, all remaining versions in `gradle.properties` are still used somewhere in the project.

This PR (so far) only aims to clean up versions from the test/example apps by using the versions listed in the `grails-bom`.
(I'm sure we can cleanup the project even further down the line.)

jdaugherty commented on examples/grails3-hibernate5/build.gradle:
```diff
@@ -5,33 +5,33 @@ dependencies {
     implementation "org.springframework.boot:spring-boot-autoconfigure"
     implementation "org.springframework.boot:spring-boot-starter-actuator"
     implementation "org.springframework.boot:spring-boot-starter-tomcat"
```
Are the other implementation statements even needed with the bom?

matrei replied:
A `bom` will not change the need to import all required dependencies.

I have not cleaned up the list of dependencies in this PR. Only removed the versions that are already provided by `grails-bom`. I am sure the dependency list can be cleaned up a bit as well.

jdaugherty commented on examples/grails3-hibernate5/build.gradle:
```diff
@@ -5,33 +5,33 @@ dependencies {
     implementation "org.springframework.boot:spring-boot-autoconfigure"
```
I'm guessing the grails gradle plugin includes the grails-bom by default?  I'm not as familiar with the way the bom's work - does it just pull in the versions or does it automatically add the dependencies too?  Are any of these implementations even needed if the bom is included?

matrei replied:
Yes, the `grails-gradle-plugin` adds the `grails-bom` to the project.

A `bom` only provides suggested versions for the modules it has listed, so you don't have to provide specific versions yourself. It does not add any dependencies by itself.

matrei committed:
chore: Remove outdated comment

matrei committed:
chore: Cleanup `examples-grails3-database-per-tenant`

- `AnotherBookService`: Make `save()` method use passed book title
- `Application`: Remove unused imports and old comment
- `application.yml`: Format properly with 2 spaces and group config settings better
- `logback.xml`: Correct faulty pattern
- `build.gradle`: Use correct dependencies and narrow the scopes. Apply Gradle plugins in subproject instead of having logic and conditions in the root build file apply them - much easier to see what's being applied.

matrei committed:
chore: Cleanup `examples-grails3-hibernate5`

- `application.yml`: Format properly with 2 spaces and group config settings better. Remove unused config.
- `logback.xml`: Correct faulty pattern
- `build.gradle`: Use correct dependencies and narrow the scopes. Switch from `tomcat-jdbc` to `HikariCP` as this will be the new default. Apply Gradle plugins in subproject instead of having logic and conditions in the root build file apply them - much easier to see what's being applied.
- `BookControllerUnitSpec`: Remove unused imports and bean definition of deprecated `MimeTypesGrailsPlugin`.
- `UrlMappings`: Move to package so it is pickup up.

matrei committed:
chore: Cleanup `examples-grails3-multiple-datasources`

- `application.yml`: Format properly with 2 spaces and group config settings better. Remove unused config.
- `logback.xml`: Correct faulty pattern
- `Application`: Remove unused imports and old comment
- `build.gradle`: Use correct dependencies and narrow the scopes. Switch from `tomcat-jdbc` to `HikariCP` as this will be the new default. Apply Gradle plugins in subproject instead of having logic and conditions in the root build file apply them - much easier to see what's being applied. Use `astTransformation` configuration for dependencies that needs to be both `compileOnly` and `integrationTestCompileOnly`.

matrei committed:
chore: Cleanup `examples-grails3-partitioned-multi-tenancy`

- `application.yml`: Format properly with 2 spaces and group config settings better. Remove unused config.
- `logback.xml`: Correct faulty pattern
- `AnotherBookService`: Make `save()` method use passed book title
- `PartionedMultiTenancySpec`: Remove unused imports
- `Application`: Remove unused imports and old comment
- `build.gradle`: Use correct dependencies and narrow the scopes. Switch from `tomcat-jdbc` to `HikariCP` as this will be the new default. Apply Gradle plugins in subproject instead of having logic and conditions in the root build file apply them - much easier to see what's being applied.

matrei committed:
chore: Cleanup `examples-grails3-schema-per-tenant`

- `application.yml`: Format properly with 2 spaces and group config settings better. Remove unused config.
- `logback.xml`: Correct faulty pattern
- `AnotherBookService`: Make `save()` method use passed book title
- `SchemaPerTenantSpec`: Remove unused imports
- `Application`: Remove unused imports and old comment
- `build.gradle`: Use correct dependencies and narrow the scopes. Switch from `tomcat-jdbc` to `HikariCP` as this will be the new default. Apply Gradle plugins in subproject instead of having logic and conditions in the root build file apply them - much easier to see what's being applied.

matrei committed:
chore: Cleanup `examples-grails-hibernate-groovy-proxy`

- `application.yml`: Format properly with 2 spaces and group config settings better. Remove unused config.
- `logback.xml`: Correct faulty pattern
- `Customer`: Add no-args constructor needed for proxying
- `ProxySpec`: Remove unused imports
- `Application`: Remove unused imports and old comment
- `build.gradle`: Use correct dependencies and narrow the scopes. Switch from `tomcat-jdbc` to `HikariCP` as this will be the new default. Apply Gradle plugins in subproject instead of having logic and conditions in the root build file apply them - much easier to see what's being applied. Change version variable name from `hibernateGroovyProxy` to `yakworksHibernateGroovyProxyVersion`.

codeconsole reviewed (changes requested):
@matrei have you tried running all these examples to see if they work as expected?

Do you know why `@EnableAutoConfiguration(exclude = DataSourceTransactionManagerAutoConfiguration)` was there and commented out and if it still needs to be excluded.  It can be excluded in application.yml if necessary.

matrei commented:
>  have you tried running all these examples to see if they work as expected?

Yes

> Do you know why @EnableAutoConfiguration(exclude = DataSourceTransactionManagerAutoConfiguration) was there and commented out and if it still needs to be excluded. It can be excluded in application.yml if necessary.

This was probably to prevent Spring Boot from configuring a transaction manager as GORM also does this and there is a problem when two are configured. It seems this is not needed anymore when using the `org.grails.plugins:hibernate5` plugin as it sets it's transaction manager as primary.

I actually had to add something like that to `examples-spring-boot-hibernate5` to make it not bomb when accessing the database from the started app: `@SpringBootApplication(exclude = HibernateJpaAutoConfiguration)`. It seems, this is not handled by default by the `gorm-hibernate5-spring-boot` plugin.

matrei committed:
chore: Cleanup `examples-spring-boot-hibernate5`

- `application.yml`: Format properly with 2 spaces remove unused config.
- `Application`: Exclude `HibernateJpaAutoConfiguration` to avoid duplicate transaction managers: java.lang.ClassCastException: class org.springframework.orm.jpa.EntityManagerHolder cannot be cast to class org.springframework.orm.hibernate5.SessionHolder (org.springframework.orm.jpa.EntityManagerHolder and org.springframework.orm.hibernate5.SessionHolder are in unnamed module of loader 'app')
- `BookController`: Add `@ReadOnly` to `books()` action to get a session for the `Book.list()` method.
- `build.gradle`: Use correct dependencies and narrow the scopes. Switch from `tomcat-jdbc` to `HikariCP` as this will be the new default. Apply Gradle plugins in subproject instead of having logic and conditions in the root build file apply them - much easier to see what's being applied. Add version variable `springBootGradlePluginVersion`.

matrei committed:
test: Extract test for issue 450 from `examples-grails-data-service`

The fix for grails/gorm-hibernate5#450 did not really fit in `examples-grails-data-service`.

This commit adds a new example project, that shows the problem and that it now works (taken from https://github.com/grails-core-issues-forks/multitenantcomposite).

matrei committed:
chore: Cleanup `examples-grails-data-service`

- `MultitenantBook.groovy`: Remove as extracted to examples-issue-450
- `application.yml`: Format properly with 2 spaces and group config settings better. Remove unused config.
- `ClassUsingAService`: Remove dependency on `grails-spring-security-rest`. No need to depend on outside plugins for this test. New test `ServiceInjectionSpec` exercising issue 202.
- `TestService`: Remove unused `transactionService` property
- `logback.xml`: Correct faulty pattern

matrei committed:
chore: Cleanup `examples-standalone-hibernate5`

Use correct dependencies and narrow the scopes. Switch from `tomcat-jdbc` to `HikariCP` as this will be the new default. Apply Gradle plugins in subproject instead of having logic and conditions in the root build file apply them - much easier to see what's being applied.

codeconsole commented:
> I actually had to add something like that to `examples-spring-boot-hibernate5` to make it not bomb when accessing the database from the started app: `@SpringBootApplication(exclude = HibernateJpaAutoConfiguration)`. It seems, this is not handled by default by the `gorm-hibernate5-spring-boot` plugin.

The correct behavior I am finding is to not have to use these at all and to reconfigure the plugins to work with the AutoConfigurations.  In most cases the AutoConfigurations do 2x the work because what happens is the AutoConfigurations run first, create the service and then Grails creates the same service with the same name.   This is terribly inefficient. 

The best fixes in these scenarios are to create the bean definition before the Autoconfiguration. When not possible, you can also create a javaconfig for a null placeholder definition so that it stops the AutoConfiguration as well.

codeconsole committed:
Merge branch '9.0.x' into matrei/use-bom-in-apps

codeconsole commented:
@matrei ?

matrei commented:
Hi 😄 ! I have and was going to to some more work in this PR. But I seem to get stuck elsewhere.
